src/workerd/io/worker.c++

```diff
@@ -2140,6 +2161,58 @@ public:
 private:
   kj::WebSocket& webSocket;
 
+  void takeHeapSnapshot(jsg::Lock& js, bool exposeInternals, bool captureNumericValue) {
+    struct Activity: public v8::ActivityControl {
+      InspectorChannelImpl& channel;
+      Activity(InspectorChannelImpl& channel) : channel(channel) {}
+
+      ControlOption ReportProgressValue(uint32_t done, uint32_t total) {
+        capnp::MallocMessageBuilder message;
+        auto event = message.initRoot<cdp::Event>();
+        auto params = event.initReportHeapSnapshotProgress();
+        params.setDone(done);
+        params.setTotal(total);
+        if (done == total) {
+          params.setFinished(true);
+        }
+        auto notification = getCdpJsonCodec().encode(event);
+        channel.sendNotification(kj::mv(notification));
+        return ControlOption::kContinue;
+      }
+    };
+
+    struct Writer: public v8::OutputStream {
+      InspectorChannelImpl& channel;
+
+      Writer(InspectorChannelImpl& channel) : channel(channel) {}
+      void EndOfStream() override {}
+
+      int GetChunkSize() override {
+        return 65536;  // big chunks == faster
```

Comment:
Hey, I recognize the inimitable style of that comment. You lifted this from node / node-heapdump, didn't you? :-)

Reply:
That particular line, yes :-) I was using FileOutputStream as a guide to implement this piece and was very amused by the comment so included it when I changed the chunk size.

Reply:
I'll be sure to add a comment pointing at history :-)